libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -503,6 +492,11 @@ void AC_PosControl::init_xy()
 
     lean_angles_to_accel_xy(_accel_target.x, _accel_target.y);
 
+    // limit acceleration using maximum lean angles
+    float angle_max = MIN(_attitude_control.get_althold_lean_angle_max_cd(), get_lean_angle_max_cd());
+    float accel_max = GRAVITY_MSS * 100.0f * tanf(ToRad(angle_max * 0.01f));
+    _accel_target.xy().limit_length(accel_max);
```

Comment:
```suggestion
    _accel_target.xy().limit_length(MIN(accel_max,_accel_max_xy_cmss));
```
This change would make it functionally the same as my original PR I think? Just so I can understand the difference.

By default angle max would be 30 deg, that gives a `accel_max` of 566. The default `WPNAV_ACCEL` is 250. So there would be a change. 

In any case this is certainly an improvement. xy I max is 1000 by default to it will no longer be setting it straight to I max. 

How about limiting the `_accel_target` just for setting I? I think in that case we would not be causing a twitch?

```
Vector3f accel_target_I = _accel_target;
accel_target_I.xy().limit_length(MIN(accel_max,_accel_max_xy_cmss));
_pid_vel_xy.set_integrator(accel_target_I - _accel_desired);
```

I did not mean to change any other behavior except the initial I value.

Reply:
>This change would make it functionally the same as my original PR I think? Just so I can understand the difference.

The problem is you limited the roll and pitch like it is a two axis vector. If you limit a roll and pitch to 45/sqrt(2) degrees in the way you get you have a lean angle of 43.7795 degrees.

When the angle is only in roll or pitch you get the same answer.

The purpose of setting the I term is to make up the difference we expect on the next loop between the output of the basic P term and the current angle. So any limiting on the I term will by definition cause a twitch.

This implementation limits the twitch to the reduction in lean angle limit. A big part of your problem was we were setting the I term to give the current lean angle but then limiting the maximum lean angle to much less than that. The I term then saturated the output way past the maximum lean angle and the result was the controller was effectively frozen until the I term unwound. This problem was then further exaggerated by a low I term that meant it took about 10 seconds before the controller came out of saturation.

It is very late here but I hope that makes sense..... I will read it again in the morning :)

Reply:
Thanks for the explanation, I think it I now get it. 

Using `lean_angles_to_accel_xy` to set I seems a little none physical. That is because it is. It is not meant to represent the physics exactly but rather be the exact inverse of `accel_to_lean_angles`. Using it to set I means that the first run of the xy PID's will give the accel that will put the vehicle at the same lean angle as it was in init (ignoring P and D). So since we limit `accel_to_lean_angles` to angle max we also need to limit the angle in  `lean_angles_to_accel_xy`  so they remain the inverse of each other. 

I think the tailsitter log happened to be the "perfect storm" to highlight this issue. With this change and a much higher I term the unwind time should go from 10 seconds to less than half a second.

Reply:
>(ignoring P and D)

P and D are both zero due to the other settings in the init. So you are correct:

>the first run of the xy PID's will give the accel that will put the vehicle at the same lean angle as it was in init 

provided the vehicle lean angle is less than the maximum lean angle.